### PR TITLE
add SEQAN_IF_CONSTEXPR

### PR DESCRIPTION
**Pull request by h-2**
- [x] wait for #2209 then rebase
  - [ ] other obvious candidates where we can use this?

### CODE REVIEW

**Comment by rrahn** [comment]
seems you broke all builds 🥇

File: include/seqan/find/find_score.h [hunk 1]
```diff
@@ -375,12 +375,15 @@ _findScoreSimpleProportional(TFinder & finder, Pattern<TNeedle, DPSearch<TScore,
     TSize haystack_length = length(container(hostIterator(finder)));
 
     //limit search width for prefix search
-    if (IsSameType<TSpec, FindPrefix>::VALUE && (score_gap < 0))
+    SEQAN_IF_CONSTEXPR (IsSameType<TSpec, FindPrefix>::VALUE)
     {
-        TSize maxlen = prefix_begin_position + length(needle(me)) + ((scoreLimit(me) - me.data_maxscore) / score_gap) + 1;
-        if (haystack_length > maxlen)
+        if (score_gap < 0))
```
Comment (rrahn) [inline include/seqan/find/find_score.h]: Here is your error 🔍

**Comment by h-2** [comment]
>  it should rather have positive effects on the performance but I strongly advice doing some micro-benchmarks. 

Well, I think there is no way this can have negative effects. It reduces code generation and branching. In the worst case it will be the same as not using it.

**Comment by h-2** [comment]
rebased on current develop with c++17 CI

**Pull request merged**